aerogear/mobile-security-service-operator
# SDKConfigValues, Validations, Uninstall scenario

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9240
https://issues.jboss.org/browse/AEROGEAR-9224

## What
- Change the SDKConfig struct as it was requested
- Move validation of cluster protocol to MSS pkg. It should not be used for another CR
- Fix finalizer for the scenario when the service was deleted (UNINSTALL)
- Rename the const in order to follow the golang convection
 
## Why
- Keep easy to be maintained
- Attend requirements. 

## Verification Steps
Use the image: `docker.io/cmacedo/mobile-security-service-operator:SDK-CONFIG`
1. Install all
2. Create an app and check that SDKConfig is as defined in the trello card. (https://trello.com/c/ExZqTlJ6/12-5-mss-consumption-cr-as-a-developer-i-want-to-be-able-to-specify-necessary-info-through-a-cr-such-that-i-can-associate-my-mobile)
4. Ensure that the app is created in the Service
5. Delete the app and ensure that the app was removed from the service 
6. Test the delete-all and check that it should not be stuck. 

## Checklist:


## Progress



<img width="988" alt="Screenshot 2019-05-23 at 10 32 22" src="https://user-images.githubusercontent.com/7708031/58242278-6a882200-7d46-11e9-9db3-f7ae94ea2bc9.png">
<img width="1321" alt="Screenshot 2019-05-23 at 10 32 15" src="https://user-images.githubusercontent.com/7708031/58242280-6a882200-7d46-11e9-9793-da4a97ca7eff.png">
